percona/controller/pgbackup/controller.go

```diff
@@ -99,6 +99,24 @@ func (r *PGBackupReconciler) Reconcile(ctx context.Context, request reconcile.Re
 			return reconcile.Result{}, errors.Wrap(err, "get PostgresCluster")
 		}
 
+		if !pgCluster.Spec.Backups.IsEnabled() {
+			if err := retry.RetryOnConflict(retry.DefaultBackoff, func() error {
+				bcp := new(v2.PerconaPGBackup)
+				if err := r.Client.Get(ctx, client.ObjectKeyFromObject(pgBackup), bcp); err != nil {
+					return errors.Wrap(err, "get PGBackup")
+				}
+
+				bcp.Status.State = v2.BackupFailed
+				bcp.Status.Error = "Backups are not enabled in the PerconaPGCluster configuration"
```

Comment:
Maybe we can add one more state like skipped or disabled? Technically it's not failed it's just disabled.